src/Microsoft.AspNet.Http.Abstractions/HttpResponse.cs

```diff
@@ -16,34 +19,90 @@ public abstract class HttpResponse
+        /// <summary>
+        /// Gets the response headers.
+        /// </summary>
         public abstract IHeaderDictionary Headers { get; }
 
+        /// <summary>
+        /// Gets the response body <see cref="Stream"/>.
+        /// </summary>
         public abstract Stream Body { get; set; }
 
+        /// <summary>
+        /// Gets or sets the length of the response. If specified, this value will be used to set the value of the
+        /// <c>Content-Length</c> header.
+        /// </summary>
         public abstract long? ContentLength { get; set; }
 
+        /// <summary>
+        /// Gets or sets the value for the <c>Content-Type</c> response header.
+        /// </summary>
         public abstract string ContentType { get; set; }
 
+        /// <summary>
+        /// Gets the collection of response cookies.
+        /// </summary>
         public abstract IResponseCookies Cookies { get; }
 
+        /// <summary>
+        /// Gets a value indicating whether response headers have been sent to the client.
+        /// </summary>
         public abstract bool HasStarted { get; }
 
+        /// <summary>
+        /// Adds a delegate to be invoked just before response headers will be sent to the client.
+        /// </summary>
+        /// <param name="callback">The delegate to execute.</param>
+        /// <param name="state">A state object to capture and pass back to the delegate.</param>
         public abstract void OnStarting(Func<object, Task> callback, object state);
 
+        /// <summary>
+        /// Adds a delegate to be invoked just before response headers will be sent to the client.
+        /// </summary>
+        /// <param name="callback">The delegate to execute.</param>
         public virtual void OnStarting(Func<Task> callback) => OnStarting(_callbackDelegate, callback);
 
+        /// <summary>
+        /// Adds a delegate to be invoked after the response has finished being sent to the client.
+        /// </summary>
+        /// <param name="callback">The delegate to invoke.</param>
+        /// <param name="state">A state object to capture and pass back to the delegate.</param>
         public abstract void OnCompleted(Func<object, Task> callback, object state);
 
+        /// <summary>
+        /// Registers an object for disposal by the host once the request has finished processing.
+        /// </summary>
+        /// <param name="disposable">The object to be disposed.</param>
         public virtual void RegisterForDispose(IDisposable disposable) => OnCompleted(_disposeDelegate, disposable);
 
+        /// <summary>
+        /// Adds a delegate to be invoked after the response has finished being sent to the client.
+        /// </summary>
+        /// <param name="callback">The delegate to invoke.</param>
         public virtual void OnCompleted(Func<Task> callback) => OnCompleted(_callbackDelegate, callback);
 
+        /// <summary>
+        /// Returns a temporary redirect response (302) to the client.
```

Comment:
I kind of like saying `HTTP 302` instead of just `302`. (To us it's obvious, but perhaps less so to some.)

This comment applies to the next API too.